module/DTO/questionsDTO.js

```diff
@@ -0,0 +1,541 @@
+          if (children.length > 0) evidenceSections.push(assessment);
+        }
+        resolve({
+          success: true,
+          message: messageConstants.apiResponses.EVIDENCE_FETCHED,
+          data: {
+            evidenceSections,
+            submissionDocumentCriterias,
+          },
+        });
+      } catch (error) {
+        return reject({
+          success: false,
+          message: error.message,
+          data: false,
+        });
+      }
+    });
+  }
+
+  /**
+   * Processes child questions for a given criteria and updates page questions with transformed data.
+   * @method
+   * @name getPageQuestions
+   * @param {Object} criteria - The criteria object.
+   * @param {Array} children - Array of child question.
+   * @param {Array} pageQuestions
+   * @returns {Promise<Array>} - Resolves with the updated array of transformed page questions.
+   */
+  static getPageQuestions(criteria, children, pageQuestions) {
+    const readQuestions = [];
+    return new Promise((resolve, reject) => {
+      const processChild = async (j) => {
+        try {
+          const res = await readQuestion(children[j]?.identifier);
+
+          if (res.body.responseCode !== httpStatusCode.ok.code) {
+            return reject({
+              message: res.params.errmsg,
+              status: httpStatusCode.bad_request.status,
+            })
+          }
+
+          let childData = res?.data;
+          readQuestions.push(childData);
+
+          // branchingLogic - Branching logic refers to the condition where, if a question has a visibleIf property, we add branching logic. This means that if certain options are selected, new questions are triggered based on the option selected.
+          const branching = criteria?.branchingLogic;
+
+          if (!isEmpty(branching) && childData) {
+            childData = await this.updateChildDataWithBranching(
+              branching,
+              childData,
+              readQuestions,
+              children
+            )?.data;
+          }
+
+          // question transformation
+          const childTemplate = await this.transformQuestionData(
+            {},
+            childData,
+            j,
+            children[j]
+          )?.data;
+
+          pageQuestions.push(childTemplate);
+        } catch (error) {
+          reject({
+            success: false,
+            message: `Error processing child at index ${j}: ${error.message}`,
+            data: false,
+          });
+        }
+      };
+
+      const processAllChildren = async () => {
+        for (let j = 0; j < children.length; j++) {
+          await processChild(j);
+        }
+        resolve({
+          success: true,
+          message: messageConstants.apiResponses.PAGE_QUESTION_FETCHED,
+          data: {
+            ...pageQuestions,
+          },
+        });
+      };
+
+      processAllChildren();
+    });
+  }
+
+  /**
+   * Retrieves the template type of a given question.
+   *
```

Comment:
@jayesh9747  please correct the function signtaure in entire code which u have witten